test/fluree/db/policy/parsing.clj

```diff
@@ -0,0 +1,116 @@
+                          [;; assign root-did to :ex/rootRole
+                           {:id     root-did
+                            :f/role :ex/rootRole}
+                           ;; assign alice-did to :ex/userRole and also link the did to :ex/alice via :ex/user
+                           {:id      alice-did
+                            :ex/user :ex/alice
+                            :f/role  :ex/userRole}
+                           {:id      customer-did
+                            :ex/user :ex/bob
+                            :f/role  :ex/customerRole}
+                           {:id           :ex/rootPolicy,
+                            :type         [:f/Policy],
+                            :f/targetNode :f/allNodes       ;; :f/allNodes special keyword meaning every node (everything)
+                            :f/allow      [{:id           :ex/rootAccessAllow
+                                            :f/targetRole :ex/rootRole
+                                            :f/action     [:f/view :f/modify]}]}
+                           {:id            :ex/UserPolicy,
+                            :type          [:f/Policy],
+                            :f/targetClass :ex/User
+                            :f/allow       [{:id           :ex/globalViewAllow
+                                             :f/targetRole :ex/userRole
+                                             :f/action     [:f/view]}
+                                            {:f/targetRole :ex/userRole
+                                             :f/action     [:f/modify]
+                                             ;; by default, user can modify their own user profile (following relationship from identity/DID -> :ex/user to User object
+                                             :f/equals     {:list [:f/$identity :ex/user]}}]
+                            :f/property    [{:f/path  :schema/ssn
+                                             :f/allow [{:id           :ex/ssnViewRule
+                                                        :f/targetRole :ex/userRole
+                                                        :f/action     [:f/view]
+                                                        :f/equals     {:list [:f/$identity :ex/user]}}]}]}])
+          sid-userRole @(fluree/promise-wrap (dbproto/-subid db :ex/userRole))
+          sid-alice    @(fluree/promise-wrap (dbproto/-subid db alice-did))
+          sid-User     @(fluree/promise-wrap (dbproto/-subid db :ex/User))
+          sid-ssn      @(fluree/promise-wrap (dbproto/-subid db :schema/ssn))
+          ;; create optimized policy map for ex:userRole
+          policy-alice @(fluree/promise-wrap (policy/policy-map db alice-did :ex/userRole nil))
+          policy-root  @(fluree/promise-wrap (policy/policy-map db root-did :ex/rootRole nil))]
+
+      ;; look at  policy for user
+      (is (= (replace-policy-fns policy-alice)              ;; replace compiled functions for data comparisons below
```

Comment:
It looks like you've swapped the `expected` and `actual` positions here, just fyi that the test runner produces more readable failures if you reverse the arguments to `=`.

Reply:
Fixed, thanks!